src/main/java/io/cryostat/agent/MainModule.java

```diff
@@ -38,52 +38,127 @@
 package io.cryostat.agent;
 
 import java.net.URI;
+import java.net.http.HttpClient;
+import java.security.KeyManagementException;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.time.Duration;
 import java.util.UUID;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.atomic.AtomicInteger;
 
 import javax.inject.Named;
 import javax.inject.Singleton;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.X509TrustManager;
 
 import dagger.Lazy;
 import dagger.Module;
 import dagger.Provides;
-import io.vertx.core.Vertx;
-import io.vertx.core.VertxOptions;
 
 @Module(includes = {ConfigModule.class})
 public abstract class MainModule {
 
     @Provides
     @Singleton
-    public static Vertx provideVertx() {
-        return Vertx.vertx(new VertxOptions().setEventLoopPoolSize(1).setWorkerPoolSize(1));
+    public static AtomicInteger provideThreadId() {
+        return new AtomicInteger(0);
     }
 
     @Provides
     @Singleton
-    public static WebServer provideHttpServer(
+    public static ScheduledExecutorService provideExecutor(AtomicInteger threadId) {
+        return Executors.newScheduledThreadPool(
+                3,
```

Comment:
It does seem arbitrary, but since this is no longer using a nice non-blocking/reactive framework like Vert.x, it's basically because it needs one thread to do each thing it might need to do in parallel. Sending an HTTP request to the Cryostat server, receiving requests from Cryostat on the built-in webserver, and then general work like processing lifecycle events etc.